### PR TITLE
chore: refactor point cloud rendering into reveal architecture 

### PR DESCRIPTION
**Pull request by Savokr**
# Description

Refactoring of Potree based point cloud rendering into reveal rendering package. Also, `PointCloudMaterialManager` was created to make managing materials for point clouds similar to how it is done with CAD models.

# Checklist:

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the public documentation. (No external changes)
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [ ] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/rendering/src/pointcloud-rendering/enums.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/*!
```
Comment (haakonflatval-cognite) [inline viewer/packages/rendering/src/pointcloud-rendering/enums.ts:1]: I don't know how I feel about adding a Copyright notice on these files when they're just copy-paste from another project :thinking:

**Review by haakonflatval-cognite — APPROVED** [review]
Looks very good! Only one comment we might want to discuss...

**Review by haakonflatval-cognite — COMMENTED** [review]
Another suggestion - I see that the `CadMaterialManager` handles the clipping planes for cad models. I'd suggest moving the clipping plane logic for point clouds from `PointCloudManager` to `PointCloudMaterialManager` as well.

File: viewer/packages/rendering/src/pointcloud-rendering/enums.ts (hunk 1 above)
Comment (Savokr) [reply viewer/packages/rendering/src/pointcloud-rendering/enums.ts:1]: Replaced with different header

**Pull request merged**